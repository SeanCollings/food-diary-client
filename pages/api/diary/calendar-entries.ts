import {
  formatMonthNumberYear,
  getDayFromDate,
  getMidnightISODaysInMonth,
  getMonthAndYearFromDate,
  getNewMonth,
  setDateMidnightISOString,
} from '@utils/date-utils';
import { NextApiRequest, NextApiResponse } from 'next';
import { diaryMealsMockData, diaryWellnessMockData } from '@client/mock';
import { URI_DIARY_CALENDAR_ENTRIES } from '@client/constants';
import { ICalendarEntriesResponseBody } from '@client/interfaces/diary-data';

interface IRequest extends NextApiRequest {
  query: {
    date: string;
    months: string;
  };
}

interface IEntriesPerMonth {
  [date: string]: number[];
}

const getDatesOfEntriesInDateRange = <T extends unknown>(
  dateRange: string[],
  entries: { [date: string]: T }
) => {
  const entryDates = dateRange.reduce((acc, date) => {
    if (entries[date]) {
      acc.push(getDayFromDate(date));
    }

    return acc;
  }, [] as number[]);

  return entryDates;
};

const getEntriesPerMonth = (monthRange: {
  [date: string]: boolean;
}): IEntriesPerMonth => {
  const entries = Object.keys(monthRange).reduce((acc, date) => {
    const [month, year] = getMonthAndYearFromDate(date);
    const datesInMonth = getMidnightISODaysInMonth(month, year);

    const mealEntryDays = getDatesOfEntriesInDateRange(
      datesInMonth,
      diaryMealsMockData
    );
    const wellnessEntryDays = getDatesOfEntriesInDateRange(
      datesInMonth,
      diaryWellnessMockData
    );

    const combinedEntryDays = mealEntryDays.concat(wellnessEntryDays);
    const uniqueDays = [...new Set(combinedEntryDays)];

    if (uniqueDays.length) {
      acc[formatMonthNumberYear(date)] = uniqueDays;
    }

    return acc;
  }, {} as IEntriesPerMonth);

  return entries;
};

const getPreviousMonthsRange = (date: string, months: number) => {
  let previousMonth = new Date(date);
  let monthRange = months;
  const formattedMonthRange: { [date: string]: boolean } = {
    [formatMonthNumberYear(date)]: true,
  };
  const monthRangeISO: { [date: string]: boolean } = {
    [setDateMidnightISOString(date)]: true,
  };

  while (monthRange > 1) {
    previousMonth = getNewMonth(previousMonth, 'previous');
    formattedMonthRange[formatMonthNumberYear(previousMonth)] = true;
    monthRangeISO[setDateMidnightISOString(previousMonth)] = true;
    monthRange -= 1;
  }

  return [formattedMonthRange, monthRangeISO];
};

const getDataFromAPI = async (date: string, months: string) => {
  const [monthRange, monthRangeISO] = getPreviousMonthsRange(
    date,
    parseInt(months)
  );
  const entries = getEntriesPerMonth(monthRangeISO);

  return Promise.resolve({ monthRange, entries });
};

const handler = async (
  req: IRequest,
  res: NextApiResponse<ICalendarEntriesResponseBody>
) => {
  const { date, months } = req.query;
  console.log(`-------- ${URI_DIARY_CALENDAR_ENTRIES} :`, date, months);

  if (req.method !== 'GET') {
    return;
  }

  const { monthRange, entries } = await getDataFromAPI(date, months);

  return res.status(200).json({ months: monthRange, entries });
};

export default handler;
