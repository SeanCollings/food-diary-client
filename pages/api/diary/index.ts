import { NextApiRequest, NextApiResponse } from 'next';
import { URI_DIARY } from '@client/constants';
import { IDiaryReponseData } from '@client/interfaces/diary-data';
import axios, { AxiosError } from 'axios';
import { getSession } from 'next-auth/react';

interface IRequest extends NextApiRequest {
  query: {
    date: string;
  };
}

const handler = async (
  req: IRequest,
  res: NextApiResponse<IDiaryReponseData>
) => {
  const session = await getSession({ req });
  const { date } = req.query;
  console.log(`-------- ${URI_DIARY} :`, date);

  if (req.method !== 'GET') {
    return res.status(500).json({ ok: false });
  }

  try {
    const { data } = await axios.get(
      `${process.env.SERVER_HOST}/diary?date=${date}`,
      {
        headers: {
          Authorization: `Bearer ${session?.accessToken}`,
        },
      }
    );

    return res.status(200).json({
      ok: true,
      entry: {
        meals: data?.meals || {},
        wellness: data?.wellness || {},
      },
    });
  } catch (err) {
    return res.status(500).json({
      ok: false,
      message: (
        err as AxiosError<{
          statusCode: number;
          message: string;
          error: string;
        }>
      ).response?.data.message,
    });
  }
};

export default handler;
