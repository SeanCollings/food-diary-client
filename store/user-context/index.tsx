import { getStructuredClone } from '@utils/get-structured-clone';
import {
  createContext,
  FC,
  ReactNode,
  useCallback,
  useContext,
  useMemo,
  useState,
  useEffect,
} from 'react';

export enum EUser {
  ID = 'id',
  NAME = 'name',
  EMAIL = 'email',
  AVATAR = 'avatar',
  SHARE_LINK = 'shareLink',
  STATS = 'stats',
  PREFERENCES = 'preferences',
}
export enum EPrefences {
  SHOW_DAY_STREAK = 'showDayStreak',
  SHOW_WEEKLY_EXERCISE = 'showWeeklyExcercise',
  SHOW_WEEKLY_WATER = 'showWeeklyWater',
  IS_PROFILE_SHARED = 'isProfileShared',
}

export interface IUserModel {
  [EUser.ID]: string;
  [EUser.NAME]: string;
  [EUser.EMAIL]: string;
  [EUser.AVATAR]?: string;
  [EUser.SHARE_LINK]?: string;
  [EUser.PREFERENCES]?: {
    [EPrefences.SHOW_DAY_STREAK]?: boolean;
    [EPrefences.SHOW_WEEKLY_EXERCISE]?: boolean;
    [EPrefences.SHOW_WEEKLY_WATER]?: boolean;
    [EPrefences.IS_PROFILE_SHARED]?: boolean;
  };
  [EUser.STATS]?: {
    dayStreak?: number;
    weeklyExercise?: string;
    weeklyWater?: number;
  };
}
export interface IPreferenceModel {}

type IPartialUserUpdate =
  | { [EUser.NAME]: string }
  | { [EUser.EMAIL]: string }
  | { [EUser.AVATAR]: string }
  | { [EUser.SHARE_LINK]: string };

type IPartialPreference =
  | { [EPrefences.SHOW_DAY_STREAK]?: boolean }
  | { [EPrefences.SHOW_WEEKLY_EXERCISE]: boolean }
  | { [EPrefences.SHOW_WEEKLY_WATER]: boolean }
  | { [EPrefences.IS_PROFILE_SHARED]: boolean };

export interface IUserContext {
  user: IUserModel | null;
  setInitialUser: (user: IUserModel) => void;
  updateUser: (updated: IPartialUserUpdate) => void;
  updatePreferences: (preferences: IPartialPreference) => void;
}

const initialState: IUserContext = {
  user: null,
  // user: {
  //   id: '',
  //   name: '',
  //   email: '',
  //   avatar: '',
  //   shareLink: '',
  //   darkMode: false,
  //   theme: {},
  //   preferences: {
  //     showDayStreak: true,
  //   },
  //   stats: {
  //     dayStreak: 5,
  //     weeklyExercise: '03:25',
  //     weeklyWater: 32,
  //   },
  // },
  setInitialUser: () => null,
  updateUser: () => null,
  updatePreferences: () => null,
};

const UserContext = createContext(initialState);

export const UserContextProvider: FC<{
  children: ReactNode;
  initialState?: Partial<IUserContext>;
}> = ({ children, initialState }) => {
  // console.log('initialState ::', initialState?.user);
  const [user, setCurrentUser] = useState<IUserModel | null>(
    initialState?.user || null
  );
  const [updatedUserFields, setUpdatedUserFields] = useState<
    IPartialUserUpdate[]
  >([]);
  const [updatedPreferences, setUpdatedPreferences] = useState<
    IPartialPreference[]
  >([]);

  useEffect(() => {
    if (!updatedUserFields.length) {
      return;
    }

    const timer = setTimeout(async () => {
      const payload = updatedUserFields.reduce((curr, update) => {
        return { ...curr, ...update };
      }, {} as IPartialUserUpdate);

      console.log('USER PATCH:', payload);

      setUpdatedUserFields([]);
    }, 500);

    return () => {
      clearTimeout(timer);
    };
  }, [updatedUserFields]);

  useEffect(() => {
    if (!updatedPreferences.length) {
      return;
    }

    const timer = setTimeout(async () => {
      const payload = updatedPreferences.reduce((curr, preference) => {
        return { ...curr, ...preference };
      }, {} as IPartialPreference);

      console.log('PREFERENCES PATCH:', payload);

      setUpdatedPreferences([]);
    }, 500);

    return () => {
      clearTimeout(timer);
    };
  }, [updatedPreferences]);

  const setInitialUser = useCallback((unitialUser: IUserModel) => {
    setCurrentUser(unitialUser);
  }, []);

  const updateUser = useCallback((update: IPartialUserUpdate) => {
    setCurrentUser((curr) => {
      if (!curr) {
        return null;
      }

      return {
        ...(getStructuredClone(curr || {}) || {}),
        ...(curr ? update : {}),
      };
    });

    setUpdatedUserFields((curr) => [...curr, update]);
  }, []);

  const updatePreferences = useCallback((preference: IPartialPreference) => {
    setCurrentUser((curr) => {
      if (!curr) {
        return null;
      }

      return {
        ...getStructuredClone(curr),
        preferences: { ...getStructuredClone(curr.preferences), ...preference },
      };
    });
    setUpdatedPreferences((curr) => [...curr, preference]);
  }, []);

  const context = useMemo(
    () => ({
      user,
      setInitialUser,
      updateUser,
      updatePreferences,
    }),
    [user, setInitialUser, updateUser, updatePreferences]
  );

  return (
    <UserContext.Provider value={context}>{children}</UserContext.Provider>
  );
};

export const useUserContext = () => useContext<IUserContext>(UserContext);
