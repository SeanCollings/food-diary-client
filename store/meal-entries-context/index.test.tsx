import { act, render, waitFor } from '@testing-library/react';
import { FC, useEffect, useRef } from 'react';
import {
  IMealEntries,
  MealEntriesContextProvider,
  initialState,
  useMealEntriesContext,
} from '.';
import { diaryService } from '@client/services/diary.service';
import { useUserContext } from '@store/user-context';

jest.mock('@client/services/diary.service');
jest.mock('@store/user-context');

const mockDate = '03-03-2023';

const TestComponent = () => <div>Test component</div>;

const DisplayEntries: FC<{
  entries: IMealEntries;
}> = ({ entries }) => (
  <>
    {Object.keys(entries).map((key) => (
      <div key={key}>
        Key: {key}
        <div>{JSON.stringify(entries[key])}</div>
      </div>
    ))}
  </>
);

const MockInitialEntries: FC<{ setEntries?: boolean }> = ({ setEntries }) => {
  const { mealEntries, requestSetMealEntries } = useMealEntriesContext();

  useEffect(() => {
    if (setEntries) {
      requestSetMealEntries({
        meals: {
          '01-02-2023': {
            snack_1: {
              contents: [
                { id: 's1_id_1', food: 's1_food_1' },
                { id: 's1_id_2', food: 's1_food_2' },
              ],
            },
            dinner: {
              contents: [{ id: 'd_id', food: 'd_food' }],
            },
          },
        },
      });
    }
  }, [requestSetMealEntries, setEntries]);

  return <DisplayEntries entries={mealEntries} />;
};

const MockRequestMealEntries: FC<{
  target: 'add' | 'update' | 'delete';
  alternate?: boolean;
}> = ({ target, alternate }) => {
  const runRef = useRef(false);
  const { mealEntries, addMealEntry, updateMealEntry, removeMealEntryById } =
    useMealEntriesContext();

  useEffect(() => {
    if (runRef.current) {
      return;
    }

    runRef.current = true;

    if (target === 'add') {
      addMealEntry({
        date: '02-02-2023',
        mealId: 'lunch',
        newValues: { id: 'new_id', food: 'new_food' },
      });
    } else if (target === 'update') {
      updateMealEntry({
        date: mockDate,
        mealId: 'breakfast',
        updatedMealId: alternate ? 'breakfast' : 'snack_2',
        updatedContent: { id: 'update_id', food: 'update_food' },
      });
    } else {
      removeMealEntryById({
        date: mockDate,
        mealId: 'dinner',
        contentId: 'delete_id',
      });
    }
  }, [target, alternate, addMealEntry, updateMealEntry, removeMealEntryById]);

  return <DisplayEntries entries={mealEntries} />;
};

describe('store - meal-entries-context', () => {
  const mockDiaryService = jest.mocked(diaryService);
  const mockUseUserContext = jest.mocked(useUserContext);
  const mockError = jest.fn();

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(mockError);
  });

  beforeEach(() => {
    mockDiaryService.createMealEntry.mockResolvedValue({} as any);
    mockDiaryService.updateMealEntry.mockResolvedValue({} as any);
    mockDiaryService.deleteMealEntry.mockResolvedValue({} as any);
    mockUseUserContext.mockReturnValue({ userLoggedIn: false } as any);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return initial state', () => {
    const {
      mealEntries,
      requestSetMealEntries,
      addMealEntry,
      updateMealEntry,
      removeMealEntryById,
    } = initialState;

    expect(mealEntries).toEqual({});
    expect(requestSetMealEntries({} as any)).toBeNull();
    expect(addMealEntry({} as any)).toBeNull();
    expect(updateMealEntry({} as any)).toBeNull();
    expect(removeMealEntryById({} as any)).toBeNull();
  });

  it('should render provider with children', () => {
    const { asFragment } = render(
      <MealEntriesContextProvider>
        <TestComponent />
      </MealEntriesContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with initial-state', () => {
    const { asFragment } = render(
      <MealEntriesContextProvider
        initialState={{
          ['02-02-2023']: {
            breakfast: {
              contents: [
                { id: 'b_1', food: 'b_food', description: 'b_description' },
              ],
            },
            lunch: {
              contents: [
                {
                  id: 'l_1',
                  food: 'l_food',
                  serving: 'l_serving',
                  quantity: 'l_quantity',
                },
              ],
            },
          },
        }}
      >
        <MockInitialEntries />
      </MealEntriesContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should handle null initial-state', () => {
    const { asFragment } = render(
      <MealEntriesContextProvider initialState={null as any}>
        <MockInitialEntries />
      </MealEntriesContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render initial entries with no initial-state', () => {
    const { asFragment } = render(
      <MealEntriesContextProvider>
        <MockInitialEntries />
      </MealEntriesContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should set requested entries', async () => {
    const { asFragment } = render(
      <MealEntriesContextProvider>
        <MockInitialEntries setEntries />
      </MealEntriesContextProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Add-a-meal', () => {
    it('should add a meal entry for a logged-in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      const { asFragment } = render(
        <MealEntriesContextProvider>
          <MockRequestMealEntries target="add" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.createMealEntry).toHaveBeenCalledWith({
          body: {
            content: { food: 'new_food', id: 'new_id' },
            mealId: 'lunch',
          },
          date: '02-02-2023',
        });
      });
      expect(asFragment()).toMatchSnapshot();
    });

    it('should handle errors in adding a meal entry for a logged-in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);
      mockDiaryService.createMealEntry.mockResolvedValue({
        error: 'mock error occurred',
      });

      render(
        <MealEntriesContextProvider>
          <MockRequestMealEntries target="add" />
        </MealEntriesContextProvider>,
      );

      await act(() => Promise.resolve());

      expect(mockDiaryService.createMealEntry).toHaveBeenCalled();
      expect(mockError).toHaveBeenCalledWith(
        'Error creating entry:',
        'mock error occurred',
      );
    });

    it('should not send request when adding a meal entry for a non logged-in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: false } as any);

      render(
        <MealEntriesContextProvider>
          <MockRequestMealEntries target="add" />
        </MealEntriesContextProvider>,
      );

      expect(mockDiaryService.createMealEntry).not.toHaveBeenCalled();
    });
  });

  describe('Update-a-meal', () => {
    const initialState = {
      [mockDate]: {
        breakfast: {
          contents: [{ id: 'update_id', food: 'new_update_food' }],
        },
      },
    };

    it('should update a meal entry for a logged-in user from one type to another', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      const { asFragment } = render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="update" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.updateMealEntry).toHaveBeenCalledWith({
          body: {
            content: {
              food: 'update_food',
              id: 'update_id',
            },
            newMealId: 'snack_2',
            oldMealId: 'breakfast',
          },
          date: '03-03-2023',
        });
      });
      expect(asFragment()).toMatchSnapshot();
    });

    it('should update a meal entry for a logged-in user from one type to another with no original contents', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      render(
        <MealEntriesContextProvider
          initialState={{
            [mockDate]: {
              breakfast: {} as any,
            },
          }}
        >
          <MockRequestMealEntries target="update" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.updateMealEntry).toHaveBeenCalledWith({
          body: {
            content: {
              food: 'update_food',
              id: 'update_id',
            },
            newMealId: 'snack_2',
            oldMealId: 'breakfast',
          },
          date: '03-03-2023',
        });
      });
    });

    it('should update a meal entry for a logged-in user for same types', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      const { asFragment } = render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="update" alternate />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.updateMealEntry).toHaveBeenCalledWith({
          body: {
            content: {
              food: 'update_food',
              id: 'update_id',
            },
            newMealId: 'breakfast',
            oldMealId: 'breakfast',
          },
          date: '03-03-2023',
        });
      });
      expect(asFragment()).toMatchSnapshot();
    });

    it('should handle errors in adding a meal entry for a logged-in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);
      mockDiaryService.updateMealEntry.mockResolvedValue({
        error: 'mock error occurred',
      });

      render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="update" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.updateMealEntry).toHaveBeenCalled();
      });
      expect(mockError).toHaveBeenCalledWith(
        'Error updating entry:',
        'mock error occurred',
      );
    });

    it('should not send request when updating a meal entry for a non logged-in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: false } as any);

      render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="update" />
        </MealEntriesContextProvider>,
      );

      expect(mockDiaryService.updateMealEntry).not.toHaveBeenCalled();
    });
  });

  describe('Delete-meal-entry', () => {
    const initialState = {
      [mockDate]: {
        dinner: {
          contents: [
            { id: 'delete_id', food: 'remove_food' },
            { id: 'other_id', food: 'd_stay_food' },
          ],
        },
        breakfast: {
          contents: [{ id: 'b_other_id', food: 'b_stay_food' }],
        },
      },
    };

    it('should remove entry for a logged in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      const { asFragment } = render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="delete" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.deleteMealEntry).toHaveBeenCalledWith({
          body: { id: 'delete_id', mealId: 'dinner' },
          date: '03-03-2023',
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });

    it('should handle remove entry for a logged in user with no original contents', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);

      render(
        <MealEntriesContextProvider
          initialState={{
            [mockDate]: {
              dinner: {} as any,
            },
          }}
        >
          <MockRequestMealEntries target="delete" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.deleteMealEntry).not.toHaveBeenCalled();
      });
    });

    it('should handle errors when removing entry for a logged in user', async () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: true } as any);
      mockDiaryService.deleteMealEntry.mockResolvedValue({
        error: 'mock error occurred',
      });

      render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="delete" />
        </MealEntriesContextProvider>,
      );

      await waitFor(() => {
        expect(mockDiaryService.deleteMealEntry).toHaveBeenCalled();
      });
      expect(mockError).toHaveBeenCalledWith(
        'Error removing entry:',
        'mock error occurred',
      );
    });

    it('should remove entry for non-logged in user', () => {
      mockUseUserContext.mockReturnValue({ userLoggedIn: false } as any);

      render(
        <MealEntriesContextProvider initialState={initialState}>
          <MockRequestMealEntries target="delete" />
        </MealEntriesContextProvider>,
      );

      expect(mockDiaryService.deleteMealEntry).not.toHaveBeenCalled();
    });
  });
});
