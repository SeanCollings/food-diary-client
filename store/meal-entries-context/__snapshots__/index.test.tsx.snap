// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store - meal-entries-context Add-a-meal should add a meal entry for a logged-in user 1`] = `
<DocumentFragment>
  <div>
    Key: 02-02-2023
    <div>
      {"lunch":{"contents":[{"id":"new_id","food":"new_food"}]}}
    </div>
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context Delete-meal-entry should remove entry for a logged in user 1`] = `
<DocumentFragment>
  <div>
    Key: 03-03-2023
    <div>
      {"dinner":{"contents":[{"id":"delete_id","food":"remove_food"},{"id":"other_id","food":"d_stay_food"}]},"breakfast":{"contents":[{"id":"b_other_id","food":"b_stay_food"}]}}
    </div>
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context Update-a-meal should update a meal entry for a logged-in user for same types 1`] = `
<DocumentFragment>
  <div>
    Key: 03-03-2023
    <div>
      {"breakfast":{"contents":[{"id":"update_id","food":"update_food"}]}}
    </div>
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context Update-a-meal should update a meal entry for a logged-in user from one type to another 1`] = `
<DocumentFragment>
  <div>
    Key: 03-03-2023
    <div>
      {"breakfast":{"contents":[]},"snack_2":{"contents":[{"id":"update_id","food":"update_food"}]}}
    </div>
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context should handle null initial-state 1`] = `<DocumentFragment />`;

exports[`store - meal-entries-context should render initial entries with no initial-state 1`] = `<DocumentFragment />`;

exports[`store - meal-entries-context should render provider with children 1`] = `
<DocumentFragment>
  <div>
    Test component
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context should render with initial-state 1`] = `
<DocumentFragment>
  <div>
    Key: 02-02-2023
    <div>
      {"breakfast":{"contents":[{"id":"b_1","food":"b_food","description":"b_description"}]},"lunch":{"contents":[{"id":"l_1","food":"l_food","quantity":"l_quantity"}]}}
    </div>
  </div>
</DocumentFragment>
`;

exports[`store - meal-entries-context should set requested entries 1`] = `
<DocumentFragment>
  <div>
    Key: 01-02-2023
    <div>
      {"snack_1":{"contents":[{"id":"s1_id_1","food":"s1_food_1"},{"id":"s1_id_2","food":"s1_food_2"}]},"dinner":{"contents":[{"id":"d_id","food":"d_food"}]}}
    </div>
  </div>
</DocumentFragment>
`;
